web/core/components/workspace/sidebar/favorites/favorites-menu.tsx

```diff
@@ -0,0 +1,168 @@
+"use client";
+
+import React, { useEffect, useRef, useState } from "react";
+import { observer } from "mobx-react";
+import { useParams } from "next/navigation";
+import { ChevronRight, FolderPlus } from "lucide-react";
+import { Disclosure, Transition } from "@headlessui/react";
+// ui
+import { IFavorite } from "@plane/types";
+import { setToast, TOAST_TYPE, Tooltip } from "@plane/ui";
+// constants
+
+// helpers
+import { cn } from "@/helpers/common.helper";
+// hooks
+import { useAppTheme } from "@/hooks/store";
+import { useFavorite } from "@/hooks/store/use-favorite";
+import useLocalStorage from "@/hooks/use-local-storage";
+import { usePlatformOS } from "@/hooks/use-platform-os";
+// plane web components
+import { FavoriteFolder } from "./favorite-folder";
+import { FavoriteItem } from "./favorite-item";
+import { NewFavoriteFolder } from "./new-fav-folder";
+export const SidebarFavoritesMenu = observer(() => {
+  //state
+  const [createNewFolder, setCreateNewFolder] = useState<boolean | string | null>(null);
+  const [isScrolled, setIsScrolled] = useState(false); // scroll animation state
+
+  // store hooks
+  const { sidebarCollapsed } = useAppTheme();
+  const { favoriteIds, favoriteMap, deleteFavorite } = useFavorite();
+  const { workspaceSlug } = useParams();
+
+  const { isMobile } = usePlatformOS();
+
+  // local storage
+  const { setValue: toggleFavoriteMenu, storedValue } = useLocalStorage<boolean>("is_favorite_menu_open", false);
+  // derived values
+  const isFavoriteMenuOpen = !!storedValue;
+  // refs
+  const containerRef = useRef<HTMLDivElement | null>(null);
+
+  const handleRemoveFromFavorites = (favorite: IFavorite) => {
+    deleteFavorite(workspaceSlug.toString(), favorite.id)
+      .then(() => {
+        setToast({
+          type: TOAST_TYPE.SUCCESS,
+          title: "Success!",
+          message: "Favorite removed successfully.",
+        });
+      })
+      .catch((err) => {
+        Object.keys(err.data).map((key) => {
+          setToast({
```

Comment:
Do we need to loop through this to show all the toasts? can it be just one toast?

Reply:
In what cases do we get multiple error messages?